baseband/vdif/header.py

```diff
@@ -208,7 +208,17 @@ def fromvalues(cls, edv=False, **kwargs):
         # Some defaults that are not done by setting properties.
         kwargs.setdefault('legacy_mode', True if edv is False else False)
         kwargs['edv'] = edv
-        return super(VDIFHeader, cls).fromvalues(edv, **kwargs)
+        time = kwargs.pop('time', None)
```

Comment:
I think we can do slightly better by just testing whether we support sample rate. E.g.,
```
if 'sample_rate' in kwargs and 'sample_rate' not in cls._properties:
    # A sample_rate is needed with a passed-in time to infer seconds, frame_nr, etc., 
    # but our header does not store the sample rate, so we do it by hand.
    time = kwargs.pop('time')
    sample_rate = kwargs.pop('sample_rate')
    self = super(VDIFHeader, cls).fromvalues(edv, **kwargs)
    self.set_time(time, sample_rate)
    return self
else:
    return super(VDIFHeader, cls).fromvalues(edv, **kwargs)
```
p.s. May need something better than a `KeyError` if `time` is not present.
p.s.2 I wondered about a new subclass `VDIFNoSampleRateHeader` which has the more complicated `fromvalues`, but a quick attempt led to a proliferation of code that did not seem worth it.

Reply:
Isn't `cls._properties` not properly set until `VDIFHeader.__new__` selects the right class to create?

Reply:
On further thought, I'm changing this to:
```
        if 'sample_rate' in self._properties and sample_rate is not None:
                self.sample_rate = sample_rate
```
In this case, `set_time` will be passed `None` if `sample_rate` is not given.  If the user already passed `frame_nr`, etc., to `kwargs`, then `set_time` will either retrieve `self.set_time`, or fail as with before.

Reply:
Ah, yes, you are right, `_properties` is not defined, as the class will still typically be `VDIFHeader`, so your solution of taking out `time` and `sample_rate` before the `super` call is the better one.

I also like your revised `if` statement above -- that way it mimics most closely what `fromvalues` would normally do.
(Only, reverse the order for speed: `is not` is faster than `in`)

Reply:
I suppose setting a time should work if the user's explicitly passing `frame_nr`, so included that as well.

Reply:
No, there is no need to extract `frame_nr`, since that will get used already if it is passed in (and every header type has that).

Reply:
But where in `set_time` is `self['frame_nr']` retrieved?  In my PR I'm specifically not popping `frame_nr`, so it does get set by the class initializer, but in instances where the user has only passed a frame number and not a sample rate, `set_time` either fails or resets `self['frame_nr'] = 0` (which is about to be caught by an assertion).  I added a test for this case in `test_vdif.py`.

(It also makes sense that self[`frame_nr`] is not decoded in `set_time`, since it's probably set to an incorrect value at that point.)

Reply:
Indeed, if a time is given, then `frame_nr` should be overridden (if given as well, which really shouldn't happen). But with the `assert` statement, that would be all OK, right? I don't think one should give a `time` and a `frame_nr` (even if in principle one could infer frame_rate from it).

Reply:
If you don't think a user should be allowed to pass a `time` and `frame_nr` (I agree that in general this is silly, but one could set a time to the nearest second, then set the frame number) I'll remove the `frame_nr` getting, and reconfigure my test to make sure doing so returns some kind of error.

Reply:
If one wants that, then one should set the `frame_nr` after the fact. I don't think we should even test for whether both are passed in - our text states that setting the time also sets `frame_nr`.